exonum-java-binding-core/rust/src/proxy/node.rs

```diff
@@ -75,43 +73,28 @@ impl NodeContext {
 /// Parameters:
 /// - `node_handle` - a native handle to the native node object
 /// - `transaction` - a transaction to submit
-/// - `message` - an array containing the transaction message
-/// - `offset` - an offset from which the message starts
-/// - `size` - a size of the message in bytes
+/// - `payload` - an array containing the transaction payload
+/// - `service_id` - an identifier of the service
 #[no_mangle]
 pub extern "system" fn Java_com_exonum_binding_service_NodeProxy_nativeSubmit(
     env: JNIEnv,
     _: JClass,
     node_handle: Handle,
-    transaction: jobject,
-    message: jbyteArray,
-    offset: jint,
-    size: jint,
+    payload: jbyteArray,
+    service_id: jint,
 ) {
     let res = panic::catch_unwind(|| {
-        assert!(offset >= 0, "Offset can't be negative");
-        assert!(size >= 0, "Size can't be negative");
-        let (offset, size) = (offset as usize, size as usize);
         let node = cast_handle::<NodeContext>(node_handle);
         unwrap_jni_verbose(
             &env,
             || -> JniResult<()> {
-                let message = env.convert_byte_array(message)?;
-                let message = message[offset..offset + size].to_vec();
-                let message = RawMessage::from_vec(message);
-                let transaction = env.new_global_ref(transaction.into())?;
-                let exec = node.executor().clone();
-                let transaction = TransactionProxy::from_global_ref(exec, transaction, message);
-                if let Err(err) = node.submit(Box::new(transaction)) {
-                    let class;
-                    if err.kind() == io::ErrorKind::Other
-                        && err.description() == VERIFY_ERROR_MESSAGE
-                    {
-                        class = INVALID_TRANSACTION_EXCEPTION;
-                    } else {
-                        class = INTERNAL_SERVER_ERROR;
-                    };
-                    env.throw_new(class, err.description())?;
+                let payload = env.convert_byte_array(payload)?;
+                let service_transaction = ServiceTransaction::from_raw_unchecked(0, payload);
+                let raw_transaction = RawTransaction::new(service_id as u16, service_transaction);
```

Comment:
@bullet-tooth , @vitvakatu 
How does the transaction id get passed?

Reply:
transaction Id should be the second param of the RawTransaction

Reply:
Fixed, done